[Docs revamp] consolidated all export related tasks in one section of howtos (#355)
dgraph-io/dgraph-docs: content/howto/exportdata/export-data.md

+++
title = "Export data from Dgraph"
keywords = "export, data, self hosted"
[menu.main]
    parent = "exportdata"
    weight = 3
+++

As an `Administrator` you can export data on all nodes, configure the Alpha server, specify the export format, export to an object store, disable HTTPS for exports, and encrypt exports

> ## Export data on all nodes

Reviewer: ```suggestion
## Export data using  the GraphQL admin  endpoint
```

> You can export all nodes by executing a GraphQL mutation on the `/admin` endpoint of an Alpha node ( `localhost:8080/admin`) using any GraphQL client.

Reviewer: ```suggestion
You can export the entire data by executing a GraphQL mutation on the `/admin` endpoint of any Alpha node.
```

**Before you begin**:

> *  Specify a list or range of IP addresses allowed to initiate admin operations such as export using the command `dgraph alpha --security whitelist <IP address>`.

Reviewer: I would remove this line. It's about securing DB access and is explained elsewhere.

Author: removed.

*  Ensure that there is sufficient space on disk to store the
export.

> This mutation triggers an export for all Alpha groups of the cluster.

Reviewer: ```suggestion
This mutation triggers the export.
```

```graphql
mutation {
  export(input: {}) {
    response {
      message
      code
    }
  }
}
```

> The export data of the group:

Reviewer: we should better explain what's going on.
Maybe 
This mutation triggers the export : the Alpha node leader of each Alpha group will write an export file for the group in the directory that was specified by the --export parameter used when launching the node.
Depending on you Dgraph configuration you may obtain several files (one per Alpha group).

We recommend that you copy the files from the alpha server nodes to a safe place when the export is finished.

Reviewer: BTW how do we know that the export is done ?

Author: Good question @MichelDiz  and @rarvikar  Please provide the steps to confirm if the export was successful.

Reviewer 2: The terminal output will show "export finished" something like this. And there is a GraphQL Query to check the status.

Author: added the response

* in the Aplha instance is stored in the Alpha.
* in every other group is stored in the Alpha leader of that group.

You need to retrieve the right export files from the Alpha instances in the
cluster. Dgraph does not copy all files to the Alpha that initiated the export.

> ## Configure Dgraph Alpha server nodes

Reviewer: I would just reference the config page and put it as a pre-req step 
" Before exporting : check that you know the export directories of the alpha server nodes, see <config ref>".

Author: Yes!!! I agree.

Author: done!!!

Each Dgraph Alpha leader for a group writes output as a gzipped file to the export

> directory specified through the `--export` flag (defaults to an **export** directory). If any of the groups fail, the entire export process is considered failed and an error is returned.

Reviewer: I would add a section "Attention point".
If one of the Alpha leader does not have enough disk space or encounters an issue, the entire export is considered failed and an error is returned"

Reviewer: Action is to verify the disk space I think

Author: done!!

{{% /notice %}}

Example of a configuration to export data:

```bash
docker run --detach --rm --name dgraph-standalone \
  --publish 8080:8080 \
  --publish 8000:8000 \
  --volume ~/exports:/dgraph/myexports \
  --env "DGRAPH_ALPHA_EXPORT=/dgraph/myexports" \
  dgraph/standalone:{{< version >}}
```

## Export data format

By default, Dgraph exports data in RDF format. Replace `<FORMAT>`with `json` or `rdf` in this GraphQL mutaion:

```graphql
mutation {
  export(input: { format: "<FORMAT>" }) {
    response {
      message
      code
    }
  }
}
```

## Export to NFS or a file path

You can override the default folder path by adding the `destination` input field to the directory where you want to export data. Replace `<PATH>` in this GraphQL mutaion with the absolute path of the directory to export data.

```graphql
mutation {
  export(input: {
    format: "<FORMAT>"
    destination: "<PATH>"
  }) {
    response {
      message
      code
    }
  }
}
```

## Export to an object store

> You can export to an S3 or MinIO object store by specifying a URL in the `destination` input field.

Reviewer: We should remove initial reference to MinIO. MinIO is just a way to store to Google or Azure and we will soon remove the need to use MinIO. The feature is to save to Azure Blob Storage and Google Cloud Storage
```suggestion
You can export to an AWS S3, Azure Blob Storage or Google Cloud Storage.
```
